osquery/tables/system/windows/chocolatey_packages.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+*  Copyright (c) 2014-present, Facebook, Inc.
+*  All rights reserved.
+*
+*  This source code is licensed under the BSD-style license found in the
+*  LICENSE file in the root directory of this source tree. An additional grant
+*  of patent rights can be found in the PATENTS file in the same directory.
+*
+*/
+
+#include <boost/filesystem.hpp>
+#include <boost/property_tree/xml_parser.hpp>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/core/process.h"
+
+namespace fs = boost::filesystem;
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace tables {
+
+Status genPackage(fs::path nuspec, Row& r) {
+  std::string content;
+  if (!readFile(nuspec, content).ok()) {
+    return Status(1, "Failed to read nuspec:" + nuspec.string());
+  }
+
+  std::stringstream ss;
```

Comment:
Place this into it's own scope:
```cpp
pt::ptree propTree;
{
  std::stringstream ss;
  ss << content;
  read_xml(ss, propTree);
}
```
You could move the `std::string` and `readFile` into this limited context too. Outside of the scope you'll allow the string and string stream to be deallocated. Such that the stringstream member buffer, string buffer, and property tree buffer do not all exists together.